### ISSUE TITLE
💅 noEmptyInterface fix not applicable for ambiant module overloading

### ISSUE THREAD

**Issue by abustany** [biomejs/biome#959]
### Environment information

```bash
CLI:
  Version:                      1.4.0
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.9.0"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "pnpm/8.10.2"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 false

Workspace:
  Open Documents:               0
```

### Rule name

lint/suspicious/noEmptyInterface

### Playground link

https://biomejs.dev/playground/?code=aQBuAHQAZQByAGYAYQBjAGUAIABTAGUAcgB2AGkAYwBlAHMAIAB7AAoAIAAgAGgAZQBsAGwAbwA6ACAAcwB0AHIAaQBuAGcAOwAKAH0ACgAKAGQAZQBjAGwAYQByAGUAIABtAG8AZAB1AGwAZQAgACIAQAByAGUAbQBpAHgALQByAHUAbgAvAG4AbwBkAGUAIgAgAHsACgAgACAAZQB4AHAAbwByAHQAIABpAG4AdABlAHIAZgBhAGMAZQAgAEEAcABwAEwAbwBhAGQAQwBvAG4AdABlAHgAdAAgAGUAeAB0AGUAbgBkAHMAIABTAGUAcgB2AGkAYwBlAHMAIAB7AH0ACgB9AAoA

### Expected result

The Typescript syntax here is commonly used to allow users to extend existing interfaces in a module. Switching the line to a type alias doesn't work.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Conaclos** [comment]
Actually this affect all intreface merging:

```ts
export interface B {
  hello: string;
}

export interface A {
  world: string
}

export interface A extends B {}
```

Altough the previous code is no so useful, because it could be simplified to:

```ts
export interface B {
  hello: string;
}

export interface A extends B {
  world: string
}
```

Thus, we must either fix the code action (for example ignoring intrefaces that merges withy another one), or make the code fix unsafe.
